dune-release.opam

```diff
@@ -18,7 +18,7 @@ license: "ISC"
 homepage: "https://github.com/tarides/dune-release"
 bug-reports: "https://github.com/tarides/dune-release/issues"
 depends: [
-  "dune" {>= "2.7"}
+  "dune" {>= "3.8"}
```

Comment:
Will this mean people will have to use an older version of dune-release for older dune?

Reply:
No, dune-release is essentially just a stand-alone binary (that just happens to often be installed in the switch thus messing with users dependencies) which only uses very basic Dune commands (pretty sure it would work on projects using Dune 1.0, though we don't have tests for that). The support for the opam subfolder doesn't require any specific support from Dune; the only reason I've bumped it is to get Dune 3.8 in the tests so the feature itself can be tested.

In theory the tests could be rewritten to not use the Dune 3.8 support and just create the opam files manually, but given updating Dune has little downside I think this approach is fair enough and can also serve as an example how to apply this to one's own projects.

Reply:
Yeah, I think that's fine.

Reply:
Also, I think that you just require a dune 3.8 binary for tests. So you can depend on 2.7, keep lang 2.7, but test-depend on dune >= 3.8 with-tests.

Reply:
That's a cool idea, although specifying this in `dune-project` is a bit tricky as it injects the `dune` dependency in a wrong way:

```
(dune (and (>= 3.8) :with-test)
```

yields

```
`"dune" {>= "2.7" & >= "3.8" & with-test}
```

which I think is totally wrong since it would then have dune as a test-only dependency. OTOH

```
(dune (or (>= 2.7) (and (>= 3.8) :with-test))
```

yields

```
"dune" {>= "2.7" & >= "2.7" | >= "3.8" & with-test}
```

which *might* be correct but depends on how strong `|` binds. The [manual](https://opam.ocaml.org/doc/Manual.html#Package-Formulas) is not very explicit about it.

Reply:
You're hitting #3431 unfortunately. As a workaround, I think that you can add an explicit bound on dune, and the test one, which should generate 2 clauses and have the correct semantics. But might not be worth it, so we can put that aside.

Reply:
Ah, I knew I have seen this issue mentioned before :) Your workaround seems to work perfectly and dune will keep the dune dependency at the right minimal version in the first bound while leaving the second one alone! Neat.

Reply:
yes, I've used that before. one thing to consider is that when you'll remove this, you need to remember to remove the dune dep otherwise it will not generate the constraint that corresponds to dune lang